src/analyzer/protocol/dns/DNS.cc

```diff
@@ -1660,6 +1704,23 @@ RecordValPtr DNS_MsgInfo::BuildEDNS_COOKIE_Val(struct EDNS_COOKIE* opt) {
     return r;
 }
 
+RecordValPtr DNS_MsgInfo::BuildTKEY_Val(struct TKEY_DATA* tkey) {
+    static auto dns_tkey = id::find_type<RecordType>("dns_tkey");
+    auto r = make_intrusive<RecordVal>(dns_tkey);
+
+    r->Assign(0, query_name);
+    r->Assign(1, answer_type);
+    r->Assign(2, tkey->alg_name);
+    r->AssignTime(3, double(tkey->inception));
```

Comment:
I want to suggest the following to be consistent with size_t below, but I realize there's more of that in `DNS.cc`.

```suggestion
    r->AssignTime(3, static_cast<double>(tkey->inception));
```

Do you have thoughts/opinions here?

Reply:
I prefer `static_cast` - just forgot to swap it over after I found out about `AssignTime` :)